### PR TITLE
I added code coverage to your build

### PR DESCRIPTION
**Pull request by yamikuronue**
Also the instructions on how to specify the test_database were wrong. You're welcome.

### CODE REVIEW

**Comment by yamikuronue** [comment]
Also, if you'd like, I can do Travis and Coveralls to get you those shiny coverage badges, but you might want to up the coverage first ;)

**Comment by julianlam** [comment]
1. Where's the `coverage` file?
2. If this requres `istanbul` as a dependency, it's not listed in `package.json`
3. Database names are subjective anyhow, whether you use `1`, `mongo`, `redis`, or `foobar` doesn't really impact the test result :smile:

**Comment by yamikuronue** [comment]
1. Coverage was added to gitignore so it doesn't check in the coverage, just reports it when you build. If you'd rather check it in, I can remove that
2. I added istanbul to the dev dependencies. It's definitely there.
3. When I used `1` I got an error in the database module (real, not the mock): `error: Encountered error while running test suite: Cannot find module './database/1`. Changing that to `mongo` made the tests run and pass.

**Comment by julianlam** [comment]
Ah there it is, missed it on my first read-through.

**Comment by BenLubar** [comment]
Example of a config that works:

```
{
        "url": "http://localhost:4567",
        "secret": "c6e41882-ac49-4977-a712-312e47eb0fa4",
        "database": "mongo",
        "port": "4567",

        "mongo": {
                "host": "172.17.0.2",
                "port": 27017
        },

        "test_database": {
                "host": "172.17.0.2",
                "port": "27017",
                "password": "",
                "database": "0"
        }
}
```

**Comment by yamikuronue** [comment]
This is what I had to use:

```
{
    "test_database": {
       "host": "127.0.0.1",
        "port": "27017",
        "password": "",
        "database": "mongo"
    }
}
```

Ben's config, pasted into my c9 instance, hangs forever (even when I increase the timeout to 20,000). If I remove everything but the test_database key, it tells me the same error I reported earlier:

`error: Encountered error while running test suite: Cannot find module './database/0'`

It's possible one of the updates since I forked changed this behavior, but as far as `git@github.com:yamikuronue/NodeBB.git` is concerned, it has to include the name of the database.

**Comment by yamikuronue** [comment]
I merged Ben's changes, did a pull into my c9 instance, and got the same exact issues as before I merged them. 

Ideally, the tests wouldn't need a live database at all, so it may be more beneficial working toward that goal anyway

**Comment by boomzillawtf** [comment]
An alternative, which I've been using (I didn't see this PR before) is to leave the `test` target alone (it's faster than also running coverage) and add this to the scripts in package.json:
```
    "coverage": "istanbul cover _mocha -- -R spec ./tests -t 10000"
```

Which you can run with `npm run-script coverage`. Also, my mongo config is this:
```
"mongo": {
	"host": "127.0.0.1",
	"port": "27017",
	"username": "",
	"password": "",
	"database": "0"
},
"test_database": {
	"host": "127.0.0.1",
	"port": "27017",
	"password": "",
	"database": "1"
}
```

File: package.json [hunk 1]
```diff
@@ -44,10 +44,9 @@
     "mongodb": "~2.1.3",
     "morgan": "^1.3.2",
     "nconf": "~0.8.2",
-    "nodebb-plugin-composer-default": "3.0.19",
+    "nodebb-plugin-composer-default": "3.0.18",
     "nodebb-plugin-dbsearch": "1.0.1",
-    "nodebb-plugin-emoji-apple": "1.0.3",
-    "nodebb-plugin-emoji-extended": "1.0.5",
+    "nodebb-plugin-emoji-extended": "1.0.3",
```
Comment (julianlam) [inline package.json]: oh dear... these lines need to be reverted :smile:

**Comment by julianlam** [comment]
Hey cool, GitHub now detects if it's a merge commit. Good for GitHub.

**Comment by barisusakli** [comment]
@yamikuronue feel free to open a PR for coveralls, it would motivate us to write more tests for sure. The same happened with codeclimate.

**Pull request merged**